### PR TITLE
[goldilocks] added the option for addtionalLabels on both dashboard and controller…

### LINKED ISSUE #567: [goldilocks] Option to add podLabels
**Is your feature request related to a problem? Please describe.**
I would be super cool to be able to add podLabels to all goldilocks components. We use the labels to define teams, components and more in our kubernetes setup, but at the moment goldilocks is just floating without "our" common labels on it.

**Describe the solution you'd like**
A set of common labels and a set of configurable podLabels in the chart.

### PR DESCRIPTION
**Pull request by brondum**
goldilocks

**Why This PR?**
The option to add additional labels to the deployments and pods.

Fixes #
#567 

**Changes**
Changes proposed in this pull request:

* New option addtionalLabels to both controller and dashboard deployments.

**Checklist:**

* [ x ] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [ x ] Any new values are backwards compatible and/or have sensible default.
* [ x ] Any new values have been added to the README for the Chart, or helm-docs has been run for the charts that support it.

### CODE REVIEW

**Comment by brondum** [comment]
@lucasreed  & @sudermanjr I need a bit of help fixing the failing tests. 
even if i remove my changes the linter still fails with the same error.

And the docs thing i have no clue :D 

Thank you in advance

File: stable/goldilocks/templates/controller-deployment.yaml [hunk 1]
```diff
@@ -26,6 +29,9 @@ spec:
         app.kubernetes.io/name: {{ include "goldilocks.name" . }}
         app.kubernetes.io/instance: {{ .Release.Name }}
         app.kubernetes.io/component: controller
+        {- if .Values.controller.deployment.additionalLabels }}
```
Comment (lucasreed) [inline stable/goldilocks/templates/controller-deployment.yaml]:
```suggestion
        {{- if .Values.controller.deployment.additionalLabels }}
```
Missing a bracket here, which may be the issue with tests.

**Review by sudermanjr — CHANGES_REQUESTED** [review]
- This looks great, thanks for the contribution! Would you mind adding some test labels to the ci test-values so that we can test this in ci?

**Comment by sudermanjr** [comment]
The failing docs check might be due to the sort order. There's a flag we pass that I can look up later

**Review by lucasreed — CHANGES_REQUESTED** [review]
👋 Thanks for the PR @brondum

I added an inline comment where I think the tests are failing for you.

Also, when updating with the `helm-docs` command please make sure you use the following command:
`helm-docs --sort-values-order=file`

File: stable/goldilocks/templates/controller-deployment.yaml (hunk 1 above)
Comment (brondum) [reply stable/goldilocks/templates/controller-deployment.yaml]: I totally missed that one :) Thanks.

**Comment by brondum** [comment]
Thank you very much for the feedback, and help with fixing the PR issues. :)

**Review by lucasreed — APPROVED** [review]
LGTM Thank you for your contribution! 🚀

**Review by lucasreed — CHANGES_REQUESTED** [review]
Ah one last thing if you don't mind: Could you also add in the `dashboard.deployment.additionalLabels` definition on the `ci/test-values.yml` file. Sorry for the back and forth

**Comment by brondum** [comment]
> Ah one last thing if you don't mind: Could you also add in the `dashboard.deployment.additionalLabels` definition on the `ci/test-values.yml` file. Sorry for the back and forth

No worries, just added a single entry in the test file, hope that is okay :)

**Review by lucasreed — APPROVED** [review]
Looks great, thanks again!

**Review by sudermanjr — APPROVED** [review]

**Pull request merged**